plugins/search/src/components/HomePageComponent/SearchBarHomePageComponent.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+
+import { SearchBarPresenter } from '../SearchBar';
+import { useNavigateToQuery } from '../util';
+
+const useStyles = makeStyles({
+  searchBar: {
+    border: '1px solid #555',
+    borderRadius: '6px',
+    fontSize: '1.5em',
+  },
+});
+
+type Props = {
+  placeholder?: string;
+};
+
+export const SearchBarHomePageComponent = ({ placeholder }: Props) => {
+  const [query, setQuery] = React.useState('');
+  const handleSearch = useNavigateToQuery();
+  const classes = useStyles();
+
+  const handleSubmit = (e: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
nitpick, but it's probably a good idea to useCallback all these callbacks, as a habit - otherwise each setQuery also perhaps leads to an extra repaint of the search bar just because its props changed (new function instances)

Reply:
That's a great callout! For the `handleSubmit` function in particular, I don't think we can actually do this, since we actually need a new function instance for each update of `query`, since there's a closure on it. Since it handles keydown events, we don't get passed the full value either, so the function has to be recreated every time.

That's at least my understanding, happy to hear your thoughts if I'm mistaken.

I implemented it for the other functions in cfc6eb6.